packages/manager/apps/web/client/app/dns-zone/history/dashboard/dashboard.controller.js

```diff
@@ -0,0 +1,143 @@
+import constants from './dashboard.constants';
+
+export default class DomainDnsZoneHistoryDashboardController {
+  constructor(
+    $translate,
+    $stateParams,
+    $state,
+    $q,
+    $document,
+    Domain,
+    Alerter,
+    DNSZoneService,
+  ) {
+    this.$translate = $translate;
+    this.$stateParams = $stateParams;
+    this.$state = $state;
+    this.Domain = Domain;
+    this.Alerter = Alerter;
+    this.DNSZoneService = DNSZoneService;
+    this.$document = $document;
+    this.$q = $q;
+
+    this.listOfDnsZonesUrls = [];
+    this.dnsEntriesForComparison = [];
+    this.dnsZoneData = null;
+    this.loadingDnsZoneData = false;
+    this.zoneName = '';
+    this.loading = true;
+    this.vizualizeDnsZoneDataPopup = false;
+    this.vizualizeDnsRestorePopup = false;
+  }
+
+  goToDiffViewer() {
+    this.$state.go('app.zone.details.zone-history.diff', {
+      selectedDates: this.dnsEntriesForComparison
+        .filter((u) => u.active)
+        .map((u) => u.date),
+      productId: this.zoneName,
+    });
+  }
+
+  closeVisualizeDnsPopup() {
+    this.vizualizeDnsZoneDataPopup = false;
+  }
+
+  checkTwoElementsAreSelected() {
+    return this.dnsEntriesForComparison.filter((u) => u.active).length === 2;
+  }
+
+  visualizeDnsDataInPopup(url) {
+    this.loadingDnsZoneData = true;
+    this.DNSZoneService.readDnsFileData(url)
+      .then((res) => {
+        this.dnsZoneData = res;
+        this.loadingDnsZoneData = false;
+      })
+      .catch(({ message }) => {
+        this.Alerter.error(
+          this.$translate.instant('dashboard_history_error', { message }),
+        );
+      });
+    this.vizualizeDnsZoneDataPopup = true;
+  }
+
+  sortedByDate() {
+    return this.listOfDnsZonesUrls.sort((a, b) => {
+      return new Date(b.creationDate) - new Date(a.creationDate);
+    });
+  }
+
+  closeDnsRestorePopup() {
+    this.vizualizeDnsRestorePopup = false;
+  }
+
+  openModalDnsRestore(date) {
+    this.chosenDateForRestoreDns = date;
+    this.vizualizeDnsRestorePopup = true;
+  }
+
+  confirmRestoreDnsAtDate(date) {
+    this.DNSZoneService.restore(this.zoneName, date)
+      .catch(({ message }) => {
+        this.Alerter.error(
+          this.$translate.instant('dashboard_history_error', { message }),
+        );
+      })
+      .finally(() => this.closeDnsRestorePopup());
+  }
+
+  downloadDnsZoneFile(url) {
+    const link = this.$document[0].createElement('a');
+    if (link.download !== undefined) {
+      link.setAttribute('href', url);
+      link.setAttribute('download', constants.DNS_FILENAME);
+      link.style = 'visibility:hidden';
+      this.$document[0].body.appendChild(link);
+      link.click();
+      this.$document[0].body.removeChild(link);
+    }
+  }
+
+  goBack() {
+    this.$state.go('app.zone.details.dashboard', this.$stateParams);
+  }
+
+  getZoneHistory(zoneName) {
```

Comment:
May be zoneId makes better meaning as I see your passing productId to this function ?
```suggestion
  getZoneHistory(zoneId) {
```